src/org/openlcb/cdi/swing/CdiPanel.java

```diff
@@ -422,11 +424,33 @@ public void runBackup() {
         }
     }
 
+    private String generateFileName() {
+        String fileName = rep.getRemoteNodeAsString();
+        if (!nodeName.isEmpty()) {
+            fileName += "-"+nodeName;
+        }
+        if (rep.getCdiRep() != null && rep.getCdiRep().getIdentification() != null) {
+            // info not present if configuration hasn't been read yet
+            fileName += "-"+rep.getCdiRep().getIdentification().getSoftwareVersion();
+        }
+        fileName += "-"+java.time.LocalDate.now();
+        fileName  += "-"+java.time.LocalTime.now().format( // use default time zone
+                        java.time.format.DateTimeFormatter.ofPattern("HH:mm:ss")
```

Comment:
I think a colon could be problematic in filenames as well (at least under windows). Using dots or just writing it together might be better.